File: decine_cpp2/src/Critica.cpp
#include "Critica.h"
#include "stringconverter.h"

namespace decine {

	Critica::Critica(UsuarioVIP *usuario, const std::string &titulo, const std::string &comentario) {
		
		this->usuario = usuario;
		this->fecha = 2014; // Asignamos la fecha actual fija
		this->titulo = titulo;
		this->comentario = comentario;

	}

	std::string Critica::obtenerDatosCritica() {

		std::string resultado = "";

		resultado += std::string("\tNombre: ") + this->usuario->getNombre() + std::string("\n");
		resultado += std::string("\tFecha: ") + StringConverterHelper::toString(this->fecha) + std::string("\n");

		resultado += std::string("\tTitulo: ") + this->titulo + std::string("\n");
		resultado += std::string("\tComentario: ") + this->comentario + std::string("\n");
		resultado += std::string("\t-------------------------\n");

		return resultado;

	}
}

File: decine_java/src/decine/Critica.java
/*  
 *  Fundamentos de Ingenieria del Software
 *  Grado en Ingeniería Informática
 * 
 *  2014 © Copyleft - All Wrongs Reserved
 *
 *  Ernesto Serrano <info@ernesto.es>
 *  
 *  
 * 
 */
package decine;

import java.util.Date;

class Critica {
    
    private UsuarioVIP usuario;
    private Date fecha;
    private String titulo;
    private String comentario;
        
    Critica(UsuarioVIP usuario, String titulo, String comentario){
        this.usuario = usuario;
        this.fecha = new Date(); // Asignamos la fecha actual
        this.titulo = titulo;
        this.comentario = comentario;
        
    }
    
    String obtenerDatosCritica() {
        
        String resultado = "";
        
        resultado += "\tNombre: " + this.usuario.getNombre() + "\n";
        resultado += "\tFecha: " + String.valueOf(this.fecha) + "\n";

        resultado += "\tTitulo: " + this.titulo + "\n";
        resultado += "\tComentario: " + this.comentario + "\n";
        resultado += "\t-------------------------\n";
        
        return resultado;
        
    }
}
